src/DataTable/index.tsx

```diff
@@ -97,6 +97,7 @@ export const DataTable = <T extends Record<string, unknown>>(
   /** Table State */
   const [editing, setEditing] = useState<number | null>(null)
   const [tableData, setData] = useState<T[]>(data)
+  const resetTable = handleFetchData ? false : true
```

Comment:
@Emmaalara I'm going to approve this PR because I know this is causing issues elsewhere. We'll ideally want to update our typing to make handleFetchData required if pagination is true. I've created a ticket https://github.com/CrossroadsCX/datatable/issues/61 to track that